### PR TITLE
refactor(components): redefine sizes interface

### PR DESCRIPTION
**Pull request by rachelbt**
fixes the sizing interface.
wording will be set according to where it is applied.
not always the same values but should makes more sense this way

currently in enums -
```ts
export enum BlockSize {
	Condensed = 'condensed',
	Normal = 'normal',
	Extended = 'extended',
}

export enum Size {
	Small = 'sm',
	Medium = 'md',
	Large = 'lg',
}
```

there’s the generic Size and the Block (height) which are interpreted differently

the reason for the difference in block is that this way imply the sizing of `normal` to be a standard while other values are derivative of it.

### CODE REVIEW

File: libs/components/src/lib/button/partials/variables.scss [hunk 1]
```diff
@@ -1,2 +1,3 @@
 $icon-gap: --button-icon-gap;
+$stacked-icon-gap: --stacked-button-icon-gap;
```
Comment (yinonov) [inline libs/components/src/lib/button/partials/variables.scss:2]: Can we adopt the private access convention of variable prefixed with underline?

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/button/README.md [hunk 2]
```diff
@@ -76,12 +76,22 @@ Set the `appearance` attribute to change the button's appearance.
 <vwc-button label='outlined' appearance='outlined'></vwc-button>
 ```
 
+## Stacked
+Set the `stacked` attribute to change the button's layout to stacked.
+- Default: `false`
+
+```html preview
+<vwc-button label='Stached' appearance='filled' icon='message-sent-line' stacked size='base-small'></vwc-button>
```
Comment (yinonov) [inline libs/components/src/lib/button/README.md:84]: Notice the typo steched

**Review by yinonov — CHANGES_REQUESTED** [review]

File: libs/components/src/lib/button/README.md (hunk 2 above)
Comment (rachelbt) [reply libs/components/src/lib/button/README.md:84]: I'm the worse with typos :(

File: libs/components/src/lib/button/README.md [hunk 3]
```diff
@@ -76,12 +76,22 @@ Set the `appearance` attribute to change the button's appearance.
 <vwc-button label='outlined' appearance='outlined'></vwc-button>
 ```
 
+## Stacked
+Set the `stacked` attribute to change the button's layout to stacked.
+- Default: `false`
+
+```html preview
+<vwc-button label='stacked' appearance='filled' icon='message-sent-line' stacked size='base-small'></vwc-button>
+<vwc-button label='stacked' appearance='filled' icon='message-sent-line' icon-trailing stacked size='base'></vwc-button>
```
Comment (yinonov) [inline libs/components/src/lib/button/README.md:85]: this is another case where _base_ value isn't true.
_base_ stands for a row height of most common components (textfield, button, etc'.)

Comment (yinonov) [reply libs/components/src/lib/button/README.md:85]: needs to be discussed as I'm yet sure how this should be resolved

File: libs/components/src/lib/button/partials/variables.scss (hunk 1 above)
Comment (rachelbt) [reply libs/components/src/lib/button/partials/variables.scss]: added an underscore

File: libs/components/src/lib/button/README.md (hunk 3 above)
Comment (rachelbt) [reply libs/components/src/lib/button/README.md:85]: Not sure I got it....

Comment (yinonov) [reply libs/components/src/lib/button/README.md:85]: the `base` keyword mean a certain row height of common controls. stack button is not the same height.
This is a problem in this new `base` convention

Comment (yinonov) [reply libs/components/src/lib/button/README.md:85]: > back to the planning table

Comment (rachelbt) [reply libs/components/src/lib/button/README.md:85]: oh! Now I get it.

Comment (yinonov) [reply libs/components/src/lib/button/README.md:85]: what do we think of 1 of the former suggestions?

**density**

possible values: `-1` | `0` | `1`
default: `0`

File: libs/components/src/lib/button/button.scss [hunk 4]
```diff
@@ -48,54 +48,41 @@
...
+      @include mixins.get-size-properties(40px, button, 8px, 16px, 20px);
+    }
+  }
+
+  &.stacked {
+    @include mixins.get-size-properties(68px, body-2-bold, 10px, 16px, 18px);
+
+    &.icon-only {
+      .icon {
+        font-size: 90px;
```
Comment (rachelbt) [inline libs/components/src/lib/button/button.scss:80]: that's a bit too big 🤣 :
image

**Review by rachelbt — COMMENTED** [review]
in the popup readme - under dismissible, the button need to get `size="condensed"`.

File: libs/components/src/lib/button/button.scss (hunk 4 above)
Comment (yinonov) [reply libs/components/src/lib/button/button.scss:80]: of course 😅 it's on purpose to notice that this should be attended.

@jshenkman

- what should be the size of icon with no label within a stack button. I imagine it should be bigger?
- is shape pill applicable on a stacked button? don't feel it is designed well

**Review by yinonov — APPROVED** [review]

**Pull request merged**